### ISSUE TITLE
No IP in backend in shared network namespaces using –net=container:

### ISSUE THREAD

**Issue by amatsoukas** [containous/traefik#756]
I have a container that is running in shared network namespaces with another container (by using the option `--net=container:another-container`)

In that case, the new container will have the same network stack (and same IP) as the old container it is being linked to.

When traefik picks it, the backend's URL is set to http://:5050, the IP is missing

A docker inspect of the new container has no values in the Network part of the output (so i assume this is why traefik does not pick the IP). 

In that case, traefik should observe the `NetworkMode` value, which in that case is:

`"NetworkMode": "container:LINKED_CONTAINER_NAME"`,

so it can get the network settings of the new container from the network settings of the `LINKED_CONTAINER_NAME`

**Comment by Nossnevs** [comment]
See also: https://github.com/containous/traefik/issues/1076

**Comment by sapk** [comment]
@Nossnevs I don't think that it is related.

I have the same issue. I think that it is related to the fact that NetworkSettings are empty when you start a container with this context.
```
        "NetworkSettings": {
            "Bridge": "",
            "SandboxID": "",
            "HairpinMode": false,
            "LinkLocalIPv6Address": "",
            "LinkLocalIPv6PrefixLen": 0,
            "Ports": {},
            "SandboxKey": "",
            "SecondaryIPAddresses": null,
            "SecondaryIPv6Addresses": null,
            "EndpointID": "",
            "Gateway": "",
            "GlobalIPv6Address": "",
            "GlobalIPv6PrefixLen": 0,
            "IPAddress": "",
            "IPPrefixLen": 0,
            "IPv6Gateway": "",
            "MacAddress": "",
            "Networks": {}
        }
```

An example of docker-compose that trigger the same bug (based on get started config): https://gist.github.com/sapk/94de150b3639692242803b9b2bdfa526

**Comment by sapk** [comment]
It could be fixed by checking if `container.NetworkSettings.NetworkMode` start with `container:` like :
https://github.com/containous/traefik/blob/d97309646483aa39ecd0db160e68b46315077a1d/provider/docker/docker.go#L600 
and chainload to look ipaddress of container id after the string `container:`.

I could try to fix this if this solution look goods to you.

**Comment by traefiker** [comment]
Closed by #2331.
